src/plugins/WebOfTrust/util/DelayedBackgroundJob.java

```diff
@@ -0,0 +1,26 @@
+package plugins.WebOfTrust.util;
+
+/**
+ * A {@link BackgroundJob} that is executed after triggers have been aggregated for some delay.
+ *
+ * @author bertm
+ */
+public interface DelayedBackgroundJob extends BackgroundJob {
+    /**
+     * Triggers the execution of the job after either aggregating triggers for at most some delay
+     * specified by the implementation or as soon as the currently running job finishes, whichever
+     * comes last. Implementations must ensure that this method is safe to invoke from any thread
+     * at any moment.
+     */
+    @Override
+    public void triggerExecution();
```

Comment:
> There's a reason the definition is a bit mathematical: it appears slightly counterintuitive but actually is useful if you think about it. 

Hmm what is the usefulness?

If you don't go for the full-blown multiple-sentence JavaDoc I suggest, I would like to at least see the "for at most some delay" changed to a "at least the default delay": Remember what the original point behind writing this was. WOT has to deal with small tasks of work which have a large processing overhead when only processing one element. So it wants to delay the processing of them for *at least* 1 minute so there is some time for more elements to pile up so it can process N elements at once instead of only 1. The whole thing is already called "*Delayed*BackgrounbJob" because its job is to delay things. So the delay is really a minimum, not a maximum. If it was a maximum, it would rather be an Executor because it would be supposed to run it ASAP.